### PR TITLE
refactor(Selects): Use PF5 Select component instead of the deprecated one across the application

### PR DESCRIPTION
**Pull request by mmelko**
`branch: jsolovjo:e2e-tests-pf5-selectors`

### CODE REVIEW

File: packages/hawtio/src/plugins/camel/rest-services/RestServices.tsx [hunk 1]
```diff
@@ -164,11 +163,10 @@ export const RestServices: React.FunctionComponent = () => {
   }
 
   const onSelectFilterType = (
-    event: ChangeEvent<Element> | MouseEvent<Element>,
-    value: string | SelectOptionObject,
-    isPlaceholder?: boolean,
+    event: ChangeEvent<Element> | MouseEvent<Element> | undefined,
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/camel/rest-services/RestServices.tsx:166]: I'd prefer `event?:` over `| undefined`

File: packages/hawtio/src/plugins/camel/rest-services/RestServices.tsx [hunk 2]
```diff
@@ -164,11 +163,10 @@ export const RestServices: React.FunctionComponent = () => {
...
+    value: string | number | undefined,
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/camel/rest-services/RestServices.tsx:167]: same here

File: packages/hawtio/src/plugins/connect/discover/Discover.tsx [hunk 3]
```diff
@@ -96,7 +100,10 @@ export const Discover: React.FunctionComponent = () => {
     return <HawtioLoadingCard message='Please wait, discovering agents...' />
   }
 
-  const selectLabel: SelectProps['onSelect'] = (_, value) => {
+  const selectLabel = (
+    event?: React.MouseEvent<Element, MouseEvent> | undefined,
+    value?: string | number | undefined,
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/connect/discover/Discover.tsx:105]: no need for `| undefined` for them

File: packages/hawtio/src/plugins/logs/Logs.tsx [hunk 4]
```diff
@@ -128,8 +132,12 @@ const LogsTable: React.FunctionComponent = () => {
     }
   }
 
-  const onLevelSelect = (event: React.MouseEvent | React.ChangeEvent, value: string | SelectOptionObject) => {
-    const checked = (event.target as HTMLInputElement).checked
+  const onLevelSelect = (
+    event: React.MouseEvent<Element, MouseEvent> | undefined,
+    value: string | number | undefined,
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/logs/Logs.tsx:137]: I'd prefer `?` over `| undefined`

File: packages/hawtio/src/plugins/logs/Logs.tsx [hunk 5]
```diff
@@ -202,17 +210,24 @@ const LogsTable: React.FunctionComponent = () => {
...
+                </MenuToggle>
+              )}
               onSelect={onLevelSelect}
             >
-              {logLevels.map((level, index) => (
-                <SelectOption key={index} value={level} />
-              ))}
+              <SelectList>
+                {logLevels.map((level, index) => (
+                  <SelectOption hasCheckbox key={index} value={level} isSelected={filters.level.includes(level)}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/logs/Logs.tsx:226]: why use `filters.level.includes()`?  is `filters.level` an array?

File: packages/hawtio/src/plugins/quartz/jobs/Jobs.tsx [hunk 6]
```diff
@@ -89,8 +94,8 @@ export const Jobs: React.FunctionComponent = () => {
 
   const onSelect =
     (target: string) =>
-    (_event: React.MouseEvent | React.ChangeEvent, value: string | SelectOptionObject, isPlaceHolder?: boolean) => {
-      setFilters(prev => ({ ...prev, [target]: isPlaceHolder ? '' : value }))
+    (_event?: React.MouseEvent<Element, MouseEvent> | undefined, value?: string | number | undefined) => {
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/quartz/jobs/Jobs.tsx:97]: no need for `| undefined` for both.

File: packages/hawtio/src/plugins/camel/endpoints/SendMessage.tsx [hunk 7]
```diff
@@ -211,18 +217,29 @@ const MessageBody: React.FunctionComponent<{
...
-              selections={selectedFormat}
+              selected={selectedFormat}
               isOpen={isDropdownOpen}
             >
-              <SelectOption label='xml' value={Language.xml} />
-              <SelectOption label='json' value={Language.json} />
-              <SelectOption label='plaintext' value={Language.plaintext} />
+              <SelectList>
+                <SelectOption label='xml' value={Language.xml}>
+                  {Language.xml}
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/plugins/camel/endpoints/SendMessage.tsx:234]: So now `SelectOption`displaying the text that is a child component.  `Language` is an enum so I don't see a problem with that. I used it to correspond with the `{selectedFormat}` in the MenuToggle

File: packages/hawtio/src/plugins/camel/endpoints/SendMessage.tsx [hunk 8]
```diff
@@ -189,10 +194,11 @@ const MessageBody: React.FunctionComponent<{
     setMessageBody(body)
     onBodyChange(body)
   }
-  const handleToggle = () => {
-    setDropdownOpen(!isDropdownOpen)
-  }
-  const handleFormatChange = (event: React.MouseEvent | React.ChangeEvent, value: string | SelectOptionObject) => {
+
+  const handleFormatChange = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/camel/endpoints/SendMessage.tsx:199]: `| undefined`

File: packages/hawtio/src/plugins/logs/Logs.tsx [hunk 9]
```diff
@@ -128,8 +132,9 @@ const LogsTable: React.FunctionComponent = () => {
     }
   }
 
-  const onLevelSelect = (event: React.MouseEvent | React.ChangeEvent, value: string | SelectOptionObject) => {
-    const checked = (event.target as HTMLInputElement).checked
+  const onLevelSelect = (event: React.MouseEvent<Element, MouseEvent> | undefined, value?: string | number) => {
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/logs/Logs.tsx:135]: still `| undefined`

File: packages/hawtio/src/plugins/quartz/jobs/Jobs.tsx [hunk 10]
```diff
@@ -88,9 +93,8 @@ export const Jobs: React.FunctionComponent = () => {
   }
 
   const onSelect =
-    (target: string) =>
-    (_event: React.MouseEvent | React.ChangeEvent, value: string | SelectOptionObject, isPlaceHolder?: boolean) => {
-      setFilters(prev => ({ ...prev, [target]: isPlaceHolder ? '' : value }))
+    (target: string) => (_event?: React.MouseEvent<Element, MouseEvent>, value?: string | number | undefined) => {
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/quartz/jobs/Jobs.tsx:96]: unnecessary `| undefined`

**Review by tadayosi — APPROVED** [review]

**Pull request merged**